app/src/main/java/uk/gov/onelogin/network/auth/AuthCodeExchange.kt

```diff
@@ -0,0 +1,91 @@
+package uk.gov.onelogin.network.auth
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.VisibleForTesting.Companion.PRIVATE
+import com.google.gson.Gson
+import dagger.hilt.android.qualifiers.ApplicationContext
+import io.ktor.client.request.post
+import io.ktor.client.statement.bodyAsText
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Url
+import io.ktor.http.contentType
+import uk.gov.onelogin.R
+import uk.gov.onelogin.network.auth.response.TokenResponse
+import uk.gov.onelogin.network.http.IHttpClient
+import uk.gov.onelogin.network.utils.IOnlineChecker
+import javax.inject.Inject
+
+class AuthCodeExchange @Inject constructor(
+    @ApplicationContext
+    private val context: Context,
+    private val httpClient: IHttpClient,
+    private val onlineChecker: IOnlineChecker
+): IAuthCodeExchange {
+    @VisibleForTesting(otherwise = PRIVATE)
+    private var redirectUrl: Url = Url(
+        context.resources.getString(
+            R.string.webBaseUrl,
+            context.resources.getString(R.string.webRedirectEndpoint)
+        )
+    )
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    private var url: Url = Url(
+        context.resources.getString(
+            R.string.apiBaseUrl,
+            context.resources.getString(R.string.tokenExchangeEndpoint)
+        )
+    )
+
+    override suspend fun exchangeCode(code: String): TokenResponse {
```

Comment:
This is an ``override` function as the class is implementing an interface. This allows us to inject a stubbed `AuthCodeExchage` in the instrumentation tests.

The function is a `suspend` as it's doing asynchronous work (a http call in this instance), same principle as an `async` funtion in JS/TS